### PR TITLE
Add Baidu Search Plugin

### PR DESCRIPTION
**Pull request by ForestLinSen**
A separate plugin to integrate the Baidu search engine into AutoGPT

### CODE REVIEW

**Comment by ntindle** [comment]
Got my Baidu account setup today so should be able to review in less than 24 hours

**Review by desojo — CHANGES_REQUESTED** [review]
You need to put your python dependencies in requirements.txt :3

```
Found module 'baidu_search' at: plugins/Auto-GPT-Plugins/src/autogpt_plugins/baidu_search/__init__.py
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/local/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/workspace/Auto-GPT/autogpt/__main__.py", line 5, in <module>
    autogpt.cli.main()
  File "/home/vscode/.local/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/home/vscode/.local/lib/python3.10/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/home/vscode/.local/lib/python3.10/site-packages/click/core.py", line 1635, in invoke
    rv = super().invoke(ctx)
  File "/home/vscode/.local/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/vscode/.local/lib/python3.10/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/home/vscode/.local/lib/python3.10/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/workspace/Auto-GPT/autogpt/cli.py", line 90, in main
    run_auto_gpt(
  File "/workspace/Auto-GPT/autogpt/main.py", line 104, in run_auto_gpt
    cfg.set_plugins(scan_plugins(cfg, cfg.debug_mode))
  File "/workspace/Auto-GPT/autogpt/plugins.py", line 230, in scan_plugins
    loaded_plugins.append(a_module())
  File "/home/vscode/.local/lib/python3.10/site-packages/abstract_singleton/__init__.py", line 14, in __call__
    cls._instances[cls] = super().__call__(*args, **kwargs)
  File "plugins/AllowUnzippedPlugins.zip/Auto-GPT-AllowUnzippedPlugins-master/src/allow_unzipped_plugins/__init__.py", line 32, in __init__
  File "plugins/AllowUnzippedPlugins.zip/Auto-GPT-AllowUnzippedPlugins-master/src/allow_unzipped_plugins/plugin_manager.py", line 34, in load_unzipped_plugins
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/workspace/Auto-GPT/plugins/Auto-GPT-Plugins/src/autogpt_plugins/baidu_search/__init__.py", line 5, in <module>
    from .baidu_search import _baidu_search
ModuleNotFoundError: No module named 'baidu_search'
```

**Comment by ForestLinSen** [comment]
> Got my Baidu account setup today so should be able to review in less than 24 hours

I think Baidu account is not necessary here, you could try to search for something on Baidu and then use the Cookie stored in your browser to set the `.env` variable

**Comment by ForestLinSen** [comment]
> You need to put your python dependencies in requirements.txt :3
> 
> ```
> Found module 'baidu_search' at: plugins/Auto-GPT-Plugins/src/autogpt_plugins/baidu_search/__init__.py
> Traceback (most recent call last):
>   File "/usr/local/lib/python3.10/runpy.py", line 196, in _run_module_as_main
>     return _run_code(code, main_globals, None,
>   File "/usr/local/lib/python3.10/runpy.py", line 86, in _run_code
>     exec(code, run_globals)
>   File "/workspace/Auto-GPT/autogpt/__main__.py", line 5, in <module>
>     autogpt.cli.main()
>   File "/home/vscode/.local/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
>     return self.main(*args, **kwargs)
>   File "/home/vscode/.local/lib/python3.10/site-packages/click/core.py", line 1055, in main
>     rv = self.invoke(ctx)
>   File "/home/vscode/.local/lib/python3.10/site-packages/click/core.py", line 1635, in invoke
>     rv = super().invoke(ctx)
>   File "/home/vscode/.local/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
>     return ctx.invoke(self.callback, **ctx.params)
>   File "/home/vscode/.local/lib/python3.10/site-packages/click/core.py", line 760, in invoke
>     return __callback(*args, **kwargs)
>   File "/home/vscode/.local/lib/python3.10/site-packages/click/decorators.py", line 26, in new_func
>     return f(get_current_context(), *args, **kwargs)
>   File "/workspace/Auto-GPT/autogpt/cli.py", line 90, in main
>     run_auto_gpt(
>   File "/workspace/Auto-GPT/autogpt/main.py", line 104, in run_auto_gpt
>     cfg.set_plugins(scan_plugins(cfg, cfg.debug_mode))
>   File "/workspace/Auto-GPT/autogpt/plugins.py", line 230, in scan_plugins
>     loaded_plugins.append(a_module())
>   File "/home/vscode/.local/lib/python3.10/site-packages/abstract_singleton/__init__.py", line 14, in __call__
>     cls._instances[cls] = super().__call__(*args, **kwargs)
>   File "plugins/AllowUnzippedPlugins.zip/Auto-GPT-AllowUnzippedPlugins-master/src/allow_unzipped_plugins/__init__.py", line 32, in __init__
>   File "plugins/AllowUnzippedPlugins.zip/Auto-GPT-AllowUnzippedPlugins-master/src/allow_unzipped_plugins/plugin_manager.py", line 34, in load_unzipped_plugins
>   File "<frozen importlib._bootstrap_external>", line 883, in exec_module
>   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
>   File "/workspace/Auto-GPT/plugins/Auto-GPT-Plugins/src/autogpt_plugins/baidu_search/__init__.py", line 5, in <module>
>     from .baidu_search import _baidu_search
> ModuleNotFoundError: No module named 'baidu_search'
> ```

Hi, `baidu_search` is not some other module but the Python file that is located in the same folder as `__init__.py`, could you make sure it exists in the same folder? Or maybe you can try to download this plugin as a zip file in this [repo](https://github.com/ForestLinSen/autogpt-plugin-baidu) to see if it works

**Comment by p-i-** [comment]
This is a mass message from the AutoGPT core team to all open Plugin-PRs

We're in the process of re-architecting:
  https://github.com/Significant-Gravitas/Auto-GPT/wiki/Architecting

Please be patient as we work to clean up the AutoGPT core.

If you already have an existing plug-in that’s affected, please message @ntindle on Discord so we can get your plugin spinning on the new 
architecture.

If you're working on a plugin, please bear in mind that one of our primary goals in re-architecting is to make life MUCH easier for you. 
We've taken aboard feedback from plugin developers. We're on it, so please give us some time!

<3 The AutoGPT Team

File: src/autogpt_plugins/baidu_search/baidu_search.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import requests
+import json
+import os
+import re
+from bs4 import BeautifulSoup
```
Comment (ntindle) [inline src/autogpt_plugins/baidu_search/baidu_search.py:5]: Can you add this to the requirements.txt so the tests pass

**Review by ntindle — CHANGES_REQUESTED** [review]

File: src/autogpt_plugins/baidu_search/baidu_search.py (hunk 1 above)
Comment (ForestLinSen) [reply src/autogpt_plugins/baidu_search/baidu_search.py:5]: Hi, I have updated the requirements and the test has passed

**Review by ntindle — APPROVED** [review]

**Pull request merged**